tweaks in README.md file - wording and ToC

tiagomoraismorgado88 opened the pull request:
**this PR does basically aim at:**
- *reviewing wording in README file*;
- *adding Table of contents to REAME file*;

---

**main reason for that does basically lie within:**
- *improving accessibility of README file;*
- *making file easier to be read;*

---

**the main motivation between such a thing does basically lie withi:**
- *allowing users to better engage with the repository, and by doing so, increasing their interest in contributing to the community*;

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/w3c/web-platform-tests/7649)
<!-- Reviewable:end -->

Commit:
tweaks in README.md file - wording and ToC

**this PR does basically aim at:**
- *reviewing wording in README file*;
- *adding Table of contents to REAME file*;

---

**main reason for that does basically lie within:**
- *improving accessibility of README file;*
- *making file easier to be read;*

---

**the main motivation between such a thing does basically lie withi:**
- *allowing users to better engage with the repository, and by doing so, increasing their interest in contributing to the community*;

w3c-bots commented:
# Build PASSED

Started: 2017-10-10 14:39:04
Finished: 2017-10-10 14:43:36





View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/19288)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/286074039)

gsnedders commented on README.md:
````diff
@@ -82,14 +101,14 @@ line syntax is:
 ```
 
-**On Windows**: You will need to preceed the prior command with
+**On Windows**: You will need to proceed the prior command with
````
That's a bogus correction; the existing text is right.

gsnedders commented on README.md:
```diff
@@ -115,18 +134,18 @@ The `wpt` command provides a frontend to a variety of tools for
 * `wpt manifest` - For updating or generating a `MANIFEST.json` test manifest
 * `wpt install` - For installing the latest release of a browser or
-  webdriver server on the local machine.
+  web driver server on the local machine.
```
should be WebDriver

gsnedders commented on README.md:
```diff
@@ -162,9 +181,9 @@ submodules, you will need to `git submodule update`, as above. If git
+fatal: No URL found for submodule path 'resources/webidl2/test/widlproc' in .gitmodules
 Failed to recurse into submodule path 'resources/webidl2'
-fatal: No url found for submodule path 'tools/html5lib' in .gitmodules
+fatal: No URL found for submodule path 'tools/html5lib' in .gitmodules
```
Both this and the above are right (because they're copying git's error messages)

gsnedders commented on README.md:
```diff
@@ -189,11 +208,11 @@ line endings, as it will cause lint errors. For git, please set
 ============
 
-By default pregenerated certificates for the web-platform.test domain
+By default generated certificates for the web-platform.test domain
```
Again existing text is right. (Potentially should be hyphenated, but the "correction" is definitely wrong.)

gsnedders commented on README.md:
```diff
@@ -189,11 +208,11 @@ line endings, as it will cause lint errors. For git, please set
 certificates for any reason it's possible to use OpenSSL when starting
 the server, or starting a test run, by providing the
-`--ssl-type=openssl` argument to the `wpt serve` or `wpt run`
+`--SSL-type=openssl` argument to the `wpt serve` or `wpt run`
```
This is wrong, given it's a command line argument….

gsnedders commented on README.md:
```diff
@@ -235,15 +254,15 @@ The master branch is automatically synced to http://w3c-test.org/.
-that modify sensitive resources (such as `.py`). The latter require
-someone with merge access to comment with "LGTM" or "w3c-test:mirror" to
+that modify sensitive resources (such as `.py`). The latter requires
+someone with merge access to comment with "LGTM" or "w3c-test: mirror" to
```
That's wrong (it's a magic string).

gsnedders commented on README.md:
```diff
@@ -235,15 +254,15 @@ The master branch is automatically synced to http://w3c-test.org/.
-Each top-level directory matches the shortname used by a standard, with
-some exceptions. (Typically the shortname is from the standard's
+Each top-level directory matches the short name used by a standard, with
+some exceptions. (Typically the short name is from the standard's
```
This should be one word, as with the existing text. (It's a W3C term-of-art.)

gsnedders commented on README.md:
```diff
@@ -296,7 +315,7 @@ Thank you!
 ---------
 
-We have a lint tool for catching common mistakes in test files. You
+We have a lint tool for catching common mistakes in text files. You
```
Existing text is right.

gsnedders commented on README.md:
```diff
@@ -305,7 +324,7 @@ your local web-platform-tests working directory like this:
 
 The lint tool is also run automatically for every submitted pull
-request, and reviewers will not merge branches with tests that have
+the request, and reviewers will not merge branches with tests that have
```
GitHub has "pull requests", not "pull the requests"…

gsnedders commented on README.md:
```diff
@@ -330,7 +349,7 @@ either:
 
 * the `tools` subdir at the root of any top-level directory in the
-  repository which contains the tests the script is meant to be used
+  the repository which contains the tests the script is meant to be used
```
Duplicate "the" added.

gsnedders commented on README.md:
```diff
@@ -339,7 +358,7 @@ the URL for a file within them.
 If you want to add a script for use with a particular set of tests but
 there isn't yet any `tools` subdir at the root of a top-level
-directory in the repository containing those tests, you can create a
+the directory in the repository containing those tests, you can create a
```
This doesn't make sense with the "the" added here?

gsnedders commented on README.md:
```diff
@@ -2,16 +2,35 @@ The web-platform-tests Project [![IRC chat](https://goo.gl/6nCIks)](http://irc.w
 confidence that they are shipping software that is compatible with other
-implementations, and that later implementations will be compatible with
-their implementations. This in turn gives Web authors/developers
+implementations and those later implementations will be compatible with
```
The existing "that" is right here, I think?

gsnedders commented on README.md:
```diff
@@ -2,16 +2,35 @@ The web-platform-tests Project [![IRC chat](https://goo.gl/6nCIks)](http://irc.w
 confidence that they can actually rely on the Web platform to deliver on
 the promise of working across browsers and devices without needing extra
-layers of abstraction to paper over the gaps left by specification
+layers of abstraction to paper over the gaps left by the specification
```
We're talking about specifications generally here, so there should definitely be no "the" here.

tiagomoraismorgado88 commented:
i guys thanks for getting in touch. i will come back to it and review the feedback as soon as i can

Commit:
reverting all changes except table of contents

**from the perspective of reviewer, the overall impression i got was:**
- *that everything was wrong except ToC*

---

**bearing this in mind:**
- *I reverted all the text changes except table of contents*

**looking forward**
**kind regards**

tiagomoraismorgado88 commented:
#reverting all changes except table of contents

**from the perspective of reviewer, the overall impression i got was:**
- *that everything was wrong except ToC*

---

**bearing this in mind:**
- *I reverted all the text changes except table of contents*

**looking forward**
**kind regards**
